### PR TITLE
[WIP] More faction camp fixes and refactoring

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Infrastructure "More faction camp fixes and refactoring"```

#### Purpose of change
During testing #26764, a lot of other minor bugs and code confusion was discovered.  Resolve some,
but not all, of those bugs and issues.

#### Describe the solution
* add some using to short overly long variable declarations
* clean up the mission_key handling by removing the map and making sure that unavailable missions can't be selected
* consolidate the 2 of separate instances of looping through the farm into a multiuse function, and then copy-paste it somewhere else because npc pointers are weird and tricky.
* rewrite the map update code - really a work in progress and not currently part of this PR
* create a pair of functions to start each mission and end each mission, thereby guaranteeing that each
NPC gets fed, trained, etc.
* NPCs consume food for the expected mission length on mission start, and then consume more 
food as necessary on mission return.

#### Describe alternatives you've considered
Breaking this up into multiple PRs for easier review and to not delay merging:
#26926 is the first two bugfixes
#26927 adds the clearcutting mission
#26957 has the cleanup of mission_key handling
#27101 has the cleanup of the farm code
#27162 has the cleanup of start mission and end mission.
#27216 has the cleanup of faction camp calorie handlings
#27217 has a refactor of the camp_recipe_batch_max() function

#### Additional context
Partial Buglist
- [x]  Farm operations display the companion as ready to return before they're ready
- [x]  First upgrade mission requires food before food is available
- [x]  Expansion upgrade missions show as available while they're being worked on
- [ ]  Planting doesn't consume seeds
- [x]  Gathering is too hard
- [x]  Menial sort completes instantly
- [x]  Logging clearcuts too early - should be 2-3 trees
- [x]  Should reserve food -- needs testing
- [ ] Vehicles can't be on a tile when a tile is updated

### CODE REVIEW

File: src/faction_camp.cpp [hunk 1]
```diff
@@ -190,458 +192,477 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
...
-                                                                ) );
-        mission_key.push( "Reset Sort Points", _( "Reset Sort Points" ), "", false );
+            mission_key.add_return( "Recover Ally from Gathering",
+                                    _( "Recover Ally from Gathering" ), "", entry, avail );
+        }
+
+        entry = string_format( _( "Notes:\n"
+                                  "Distribute food to your follower and fill you larders.  Place "
+                                  "the food you wish to distribute opposite the tent door between "
+                                  "the manager and wall.\n \n"
```
Comment (I-am-Erk) [inline src/faction_camp.cpp]: Perhaps this text should mention that this is the default food distribution point, but might be changed by sort points.

File: src/faction_camp.cpp [hunk 2]
```diff
@@ -190,458 +192,477 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
...
-        mission_key.push( "Cut Logs", _( "Cut Logs" ), "", false, npc_list.empty() );
+        comp_list npc_list = companion_list( p, "_faction_camp_cut_log" );
+        entry = string_format( _( "Notes:\n"
+                                  "Send a companion to a nearby forest to cut logs.\n \n"
+                                  "Skill used: fabrication\n"
+                                  "Difficulty: 1 \n"
+                                  "Effects:\n"
+                                  "> 50%% of trees/trunks at the forest position will be cut down.\n"
+                                  "> 100%% of total material will be brought back.\n"
+                                  "> Repeatable with diminishing returns.\n \n"
```
Comment (I-am-Erk) [inline src/faction_camp.cpp]: Might be worth mentioning here that the forest will be cleared to a field eventually? could copy some of the text from clearcut.

File: src/faction_camp.cpp [hunk 3]
```diff
@@ -190,458 +192,477 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
...
+            entry = _( "Notes:\n"
+                       "Have a companion attempt to completely dissemble a vehicle into "
+                       "components.\n \n"
+                       "Skill used: mechanics\n"
+                       "Difficulty: 2 \n"
+                       "Effects:\n"
+                       "> Removed parts placed on the furniture in the garage.\n"
+                       "> Skill plays a huge role to determine what is salvaged.\n \n"
+                       "Risk: None\n"
+                       "Time: Skill Based \n" );
```
Comment (I-am-Erk) [inline src/faction_camp.cpp]: Time isn't skill based anymore is it?

File: src/faction_camp.cpp [hunk 4]
```diff
@@ -3141,16 +2970,18 @@ int basecamp::recruit_evaluation() const
...
     int sexpansions;
     int sfaction;
     int sbonus;
     int total = recruit_evaluation( sbase, sexpansions, sfaction, sbonus );
     std::string desc = string_format( _( "Notes:\n"
-                                         "Recruiting additional followers is very dangerous and expensive.  The outcome is heavily dependent on the skill of the "
-                                         "companion you send and the appeal of your base.\n \n"
+                                         "Recruiting additional followers is very dangerous and "
+                                         "expensive.  The outcome is heavily dependent on the skill "
+                                         "of the  companion you send and the appeal of your "
```
Comment (I-am-Erk) [inline src/faction_camp.cpp]: `the  companion`
one too many spaces here

File: src/faction_camp.cpp (hunk 3 above)
Comment (I-am-Erk) [reply src/faction_camp.cpp]: Maybe just comment out the line entirely until it's true?

File: src/faction_camp.cpp (hunk 1 above)
Comment (mlangsdorf) [reply src/faction_camp.cpp]: Fixed.

File: src/faction_camp.cpp (hunk 2 above)
Comment (mlangsdorf) [reply src/faction_camp.cpp]: Fixed.

File: src/faction_camp.cpp (hunk 3 above)
Comment (mlangsdorf) [reply src/faction_camp.cpp]: Time is 5 days fixed for now and the message says that.

Comment (mlangsdorf) [reply src/faction_camp.cpp]: Oops, I fixed this in a separate PR.

**Comment by mlangsdorf** [comment]
All the relevant commits have been merged.  There's still a zilliion bugs and issues with base camps, but this PR has served its purpose.

**Pull request closed without merging**